test/e2e/ibc_transfer_e2e_test.go

```diff
@@ -39,16 +41,132 @@ func (s *IBCTransferTestSuite) TearDownSuite() {
 	}
 }
 
+func getFirstIBCDenom(balance sdk.Coins) string {
+	// Look up the ugly IBC denom
+	denoms := balance.Denoms()
+	var denomB string
+	for _, d := range denoms {
+		if strings.HasPrefix(d, "ibc/") {
+			denomB = d
+			break
+		}
+	}
+	return denomB
+}
+
 func (s *IBCTransferTestSuite) Test1IBCTransfer() {
-	bbnChain := s.configurer.GetChainConfig(0)
+	denom := "ubbn"
+	amount := int64(1_000_000)
+	delta := float64(10000) // Tolerance to account for gas fees
+
+	transferCoin := sdk.NewInt64Coin(denom, amount)
+
+	bbnChainA := s.configurer.GetChainConfig(0)
+	bbnChainB := s.configurer.GetChainConfig(1)
+
+	babylonNodeA, err := bbnChainA.GetNodeAtIndex(2)
+	s.NoError(err)
+	babylonNodeB, err := bbnChainB.GetNodeAtIndex(2)
+	s.NoError(err)
+
+	val := initialization.ValidatorWalletName
+
+	// Check balance of val in chain-A (Node 3)
+	addrA := babylonNodeA.GetWallet(val)
+	balanceA, err := babylonNodeA.QueryBalances(addrA)
+	s.Require().NoError(err)
+	// Confirm val on A has enough funds
+	s.Assert().GreaterOrEqual(balanceA.AmountOf(denom).Int64(), amount)
+
+	addrB := babylonNodeB.GetWallet(val)
+	balanceB, err := babylonNodeB.QueryBalances(addrB)
+	s.Require().NoError(err)
+	// Only one denom in B
+	s.Require().Len(balanceB, 1)
 
-	babylonNode, err := bbnChain.GetNodeAtIndex(2)
+	// Send transfer from val in chain-A (Node 3) to val in chain-B (Node 3)
+	babylonNodeA.SendIBCTransfer(val, addrB, "transfer", transferCoin)
+
+	s.Require().Eventually(func() bool {
+		// Check that the transfer is successful.
+		// Amounts have been discounted from val in chain-A and added (as a wrapped denom) to val in chain-B
+		balanceA2, err := babylonNodeA.QueryBalances(addrA)
+		if err != nil {
+			return false
+		}
+		return math.Abs(float64(balanceA.Sub(transferCoin).AmountOf(denom).Int64()-
+			balanceA2.AmountOf(denom).Int64())) < delta
+	}, 10*time.Second, 1*time.Second, "Transfer was not successful")
+
+	s.Require().Eventually(func() bool {
+		balanceB2, err := babylonNodeB.QueryBalances(addrB)
+		if err != nil {
+			return false
+		}
+		// Check that there are now two denoms in B
+		if len(balanceB2) != 2 {
+			return false
+		}
+		denomB := getFirstIBCDenom(balanceB2)
+		// Check the balance of the IBC denom
+		return math.Abs(float64(balanceB2.AmountOf(denomB).Int64()-
+			transferCoin.Amount.Int64())) < delta
```

Comment:
Does it also makes sense to ensure the suffix of `denomB` is same as `denom` in chainA?

Reply:
Checked IBC doc and it seems no :( the denom will be `ibc/XXX` where `XXX` is the has of the path but not the denom in A. Nvm